saleor/warehouse/migrations/0003_warehouse_slug.py

```diff
@@ -17,12 +17,13 @@ def create_unique_slug_for_warehouses(apps, schema_editor):
         first_char = warehouse.name[0].lower()
         if first_char != previous_char:
```

Comment:
what if some warehouse has same first letter name ?? (same for ProductType and collection)

Reply:
The idea here is to not fetch all objects of the specific type in one request. So instances are sorted by lower name, and then for every new first letter, we get slug values from objects, which slugs start with this letter.
So if some warehouses have the same first letter, new `slug_values` will not be generated. Previously generated slug data will be used in `generate_unique_slug`.